src/overlays/actors/ovl_Obj_Etcetera/z_obj_etcetera.c

```diff
@@ -26,35 +32,280 @@ const ActorInit Obj_Etcetera_InitVars = {
+    }
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex)) {
+        this->dyna.actor.objBankIndex = this->objIndex;
+        Actor_SetObjectSegment(globalCtx, &this->dyna.actor);
+        BcCheck3_BgActorInit(&this->dyna, 1);
+        thisCollisionHeader = collisionHeaders[type];
+        if (thisCollisionHeader != 0) {
+            BgCheck_RelocateMeshHeader(thisCollisionHeader, &colHeader);
+        }
+        this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, colHeader);
+        type = OBJETCETERA_TYPE(&this->dyna.actor);
+        switch (type) {
+            case TYPE_PINK_FLOWER:
+            case TYPE_PINK_FLOWER_SPAWNED_FROM_MAD_SCRUB:
+                SkelAnime_Init(globalCtx, &this->skelAnime, &D_04011518, &D_0400EB7C, this->jointTable,
+                               this->morphTable, 11);
+                this->displayListPtr = &D_0400ED80;
+                break;
+            case TYPE_GOLD_FLOWER:
+            case TYPE_GOLD_FLOWER_SPAWNED_FROM_MAD_SCRUB:
+                this->displayListPtr = &D_04011BD0;
+                SkelAnime_Init(globalCtx, &this->skelAnime, &D_040127E8, &D_0400EB7C, this->jointTable,
+                               this->morphTable, 11);
+                this->collider.dim.height = 20;
+                break;
+        }
+        type = OBJETCETERA_TYPE(&this->dyna.actor);
+        switch (type) {
+            case TYPE_PINK_FLOWER:
+            case TYPE_GOLD_FLOWER:
+                this->dyna.actor.draw = ObjEtcetera_DrawIdle;
+                this->actionFunc = ObjEtcetera_Idle;
+                Actor_SetScale(&this->dyna.actor, 0.01f);
+                this->dyna.actor.scale.y = 0.02f;
+                this->dyna.actor.focus.pos.y = this->dyna.actor.home.pos.y + 10.0f;
+                this->dyna.actor.targetMode = 3;
+                break;
+            case TYPE_PINK_FLOWER_SPAWNED_FROM_MAD_SCRUB:
+            case TYPE_GOLD_FLOWER_SPAWNED_FROM_MAD_SCRUB:
+                SkelAnime_ChangeAnim(&this->skelAnime, &D_0400EB7C, 1.0f, 0.0f,
+                                     SkelAnime_GetFrameCount(&D_0400EB7C.common), 2, 0.0f);
+                this->dyna.actor.draw = ObjEtcetera_DrawAnimated;
+                this->actionFunc = ObjEtcetera_DoIntenseOscillation;
+                Actor_SetScale(&this->dyna.actor, 0.0f);
+                this->oscillationTimer = 30;
+                this->intenseOscillationScale = 0.0f;
+                this->dyna.actor.focus.pos.y = this->dyna.actor.home.pos.y + 10.0f;
+                this->dyna.actor.targetMode = 3;
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Etcetera/func_80A7BE8C.s")
+void ObjEtcetera_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjEtcetera* this = THIS;
+    CollisionPoly* floorPoly;
+    u8 floorBgId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Etcetera/func_80A7BF08.s")
+    floorBgId = this->dyna.actor.floorBgId;
```

Comment:
You can move this definition into one line:
`u8 floorBgId = this->dyna.actor.floorBgId;`